dlt/destinations/impl/dremio/dremio.py

```diff
@@ -0,0 +1,251 @@
+
+        # extract and prepare some vars
+        bucket_path = (
+            NewReferenceJob.resolve_reference(file_path)
+            if NewReferenceJob.is_reference_job(file_path)
+            else ""
+        )
+
+        if not bucket_path:
+            raise RuntimeError("Could not resolve bucket path.")
+
+        file_name = (
+            FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+        )
+
+        bucket_url = urlparse(bucket_path)
+        bucket_scheme = bucket_url.scheme
+        if bucket_scheme == "s3" and stage_name:
+            from_clause = (
+                f"FROM '@{stage_name}/{bucket_url.hostname}/{bucket_url.path.lstrip('/')}'"
+            )
+        else:
+            raise LoadJobTerminalException(
+                file_path, "Only s3 staging currently supported in Dremio destination"
+            )
+
+        source_format = file_name.split(".")[-1]
+
+        client.execute_sql(f"""COPY INTO {qualified_table_name}
+            {from_clause}
+            FILE_FORMAT '{source_format}'
+            """)
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class DremioClient(SqlJobClientWithStaging, SupportsStagingDestination):
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: DremioClientConfiguration) -> None:
+        sql_client = DremioSqlClient(config.normalize_dataset_name(schema), config.credentials)
+        super().__init__(schema, config, sql_client)
+        self.config: DremioClientConfiguration = config
+        self.sql_client: DremioSqlClient = sql_client  # type: ignore
+        self.type_mapper = DremioTypeMapper(self.capabilities)
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        job = super().start_file_load(table, file_path, load_id)
+
+        if not job:
+            job = DremioLoadJob(
+                file_path=file_path,
+                table_name=table["name"],
+                client=self.sql_client,
+                stage_name=self.config.staging_data_source,
+                staging_credentials=(
+                    self.config.staging_config.credentials if self.config.staging_config else None
+                ),
+            )
+        return job
+
+    def restore_file_load(self, file_path: str) -> LoadJob:
+        return EmptyLoadJob.from_file_path(file_path, "completed")
+
+    def _get_table_update_sql(
+        self,
+        table_name: str,
+        new_columns: Sequence[TColumnSchema],
+        generate_alter: bool,
+        separate_alters: bool = False,
+    ) -> List[str]:
+        sql = super()._get_table_update_sql(table_name, new_columns, generate_alter)
+
+        if not generate_alter:
+            partition_list = [
+                self.capabilities.escape_identifier(c["name"])
```

Comment:
I had a look at this, but the signature of `get_columns_names_with_prop` takes a `TTableSchema` rather than ` Sequence[TColumnSchema]`, so not sure it would work here.